### ISSUE TITLE
num_classes fixed to 1000 on all tiny_vit_xm_224() models

### ISSUE THREAD

**Issue by ltong1130ztr** [microsoft/Cream#154]
Hi, 

I was trying to use tiny_vit_5m_224() for some downstream classification task with the number of classes != 1000, then I noticed that the following functions always return a tiny_vit model that outputs 1000 logits in the linear classification head:
tiny_vit_5m_224()
tiny_vit_11m_224()
tiny_vit_21m_224()
tiny_vit_21m_384()
tiny_vit_21m_512()

This seems to be caused by the wrapper register_tiny_vit_model(). Perhaps adding a few lines to overwrite the classification head is good enough to fix this?

```
def register_tiny_vit_model(fn):
    '''Register a TinyViT model
    It is a wrapper of `register_model` with loading the pretrained checkpoint.
    '''
    def fn_wrapper(pretrained=False, **kwargs):
        model = fn()
        if pretrained:
            model_name = fn.__name__
            assert model_name in _provided_checkpoints, \
                f'Sorry that the checkpoint `{model_name}` is not provided yet.'
            url = _checkpoint_url_format.format(
                _provided_checkpoints[model_name])
            checkpoint = torch.hub.load_state_dict_from_url(
                url=url,
                map_location='cpu', check_hash=False,
            )
            model.load_state_dict(checkpoint['model'])
        # overwrite the default number of classes
        if kwargs["num_classes"] != 1000:
            model.head = nn.Linear(
            model.head.in_features, kwargs["num_classes"])

        return model

    # rename the name of fn_wrapper
    fn_wrapper.__name__ = fn.__name__
    return register_model(fn_wrapper)
```

**Comment by wkcn** [comment]
Thanks for your attention to our work!

I have fixed the issue in the PR #155 .

The latest code supports building classification head with non-1000 logits. : )
